pkg/controller/account/ec2.go

```diff
@@ -42,22 +48,35 @@ func (r *ReconcileAccount) InitializeSupportedRegions(reqLogger logr.Logger, acc
 	customerTags := r.getCustomTags(reqLogger, account)
 
 	// Create go routines to initialize regions in parallel
-
 	for _, region := range regions {
 		go r.InitializeRegion(reqLogger, account, region.Name, regionAMIs[region.Name], vCPUQuota, ec2Notifications, ec2Errors, creds, managedTags, customerTags) //nolint:errcheck // Unable to do anything with the returned error
 	}
 
+	regionInitFailed := false
 	// Wait for all go routines to send a message or error to notify that the region initialization has finished
 	for i := 0; i < len(regions); i++ {
 		select {
 		case msg := <-ec2Notifications:
 			reqLogger.Info(msg)
 		case errMsg := <-ec2Errors:
-			reqLogger.Error(errors.New(errMsg), errMsg)
+			regionInitFailed = true
+			// If we fail to initialize the desired region we want to fail the account
+			reqLogger.Error(errors.New(errMsg.ErrorMsg), errMsg.ErrorMsg)
+			err := utils.SetAccountStatus(
+				r.Client,
+				reqLogger,
+				account,
+				fmt.Sprintf("Account %s failed to initialize expected region %s", account.Name, errMsg.Region),
+				awsv1alpha1.AccountInitializingRegions,
+			)
```

Comment:
I am concerned that we'll have a problem if we need to call this method twice (that is, for example, when `i=0` and `i=2`) then we may encounter an error from Kubernetes where we are trying to update an old object.

Some options could be to reload the object, or to collect all errors and update once, outside of the loop to reflect all of the errors encountered.

/hold

Reply:
Latest PR fixes this issue